crates/sui-config/src/node.rs

```diff
@@ -341,19 +345,27 @@ pub struct MetricsConfig {
 
 #[derive(Default, Debug, Clone, Deserialize, Serialize)]
 #[serde(rename_all = "kebab-case")]
-pub struct StateSnapshotConfig {
-    pub enabled: bool,
+pub struct DBCheckpointConfig {
+    #[serde(default)]
+    pub perform_db_checkpoints_at_epoch_end: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub checkpoint_path: Option<PathBuf>,
 }
 
-impl StateSnapshotConfig {
+impl DBCheckpointConfig {
     pub fn validator_config() -> Self {
-        Self { enabled: false }
+        Self {
+            perform_db_checkpoints_at_epoch_end: false,
+            checkpoint_path: None,
+        }
     }
     pub fn fullnode_config() -> Self {
-        Self { enabled: true }
+        Self {
+            perform_db_checkpoints_at_epoch_end: true,
```

Comment:
I don't think this should be default to true on fullnode config We should only need to pick maybe 1 or 2 fullnodes to enable this. What do you think?

Reply:
This makes sense, we should this be an explicit decision vs a default one. Good call. I'll make the change